### PR TITLE
feature(EJ2-67546): Upgrade packages to Latest version and Integrate Syncfusion Components

### PR DESCRIPTION
**Pull request by manikandan-mariappan**
Feature Description: 
To upgrade syncfusion packages and to use Syncfusion components wherever applicable

Solution Description:
Update all packages to latest version and Integrate following Syncfusion Components
- AppBar
- Avatar
- Card
- Skeleton

Output Screenshots:

**Before AppBar, Avatar and Card**

Screenshot (54)

**After AppBar, Avatar and Card**

Screenshot (55)

**Implemention of Skeleton Component**

Screenshot (56)

**Responsive View**

Screenshot (56)

Screenshot (58)

Screenshot (57)

### CODE REVIEW

File: src/app/app.ts [hunk 1]
```diff
@@ -22,6 +23,11 @@ export interface MyWindow extends Window {
 }
 
 
+let appbarObj = new AppBar({
```
Comment (NavinVinayagam) [inline src/app/app.ts]: Define the appbar below global variables

File: src/app/app.ts [hunk 2]
```diff
@@ -1174,7 +1180,6 @@ function InitializeWaterComponent(): void {
     linechartObj.appendTo('#balance');
     linechartObj.refresh();
 }
-
```
Comment (NavinVinayagam) [inline src/app/app.ts]: Don't remove spaces, line break, unnecessarily

File: src/app/app.ts [hunk 3]
```diff
@@ -1223,12 +1228,12 @@ function waterclick(): void {
...
     document.getElementById("food-img").style.color = '#999393';
-    document.getElementById("title").innerHTML = 'Water Consumption <span id="watertitle-annotation1">Daily Average</span><span id="watertitle-annotation2">Target</span>';
-    document.getElementById("subtitle").innerHTML = '<span id="watersubtitle-annotation1">4.32 litres</span><span id="watersubtitle-annotation2">7 litres</span>'
-    document.getElementById("watersubtitle-annotation1").style.color = '#3B61E9';
-    document.getElementById("watersubtitle-annotation2").style.color = '#3B61E9';
+    document.getElementById('chart-header-title').innerHTML = '<p class="steps-ch-title" id="title"> Water Consumption </p>'
+    document.getElementById('total-value').innerHTML = '   <p class="water-total-title" id="title"> Target </p> <p class="water-st-today subtitle" id="chart-water-total-subtitle"> 7 litres </p>'
+    document.getElementById("average-value").innerHTML = '<p class="water-daily-avg-title" id="title"> Daily Average </p> <p class="water-st-daily-average subtitle" id="chart-water-average-subtitle"> 4.32 litres </p>'
+    document.getElementById('chart-water-total-subtitle').style.color = '#3B61E9'
```
Comment (NavinVinayagam) [inline src/app/app.ts]: Element is added only in the previous line, directly add the style there.
Also, update the class, id names to simpler ones and avoid generic names

File: src/app/app.ts [hunk 4]
```diff
@@ -1261,11 +1266,12 @@ function stepclick(): void {
     document.getElementById("sleep-img").style.color = '#999393';
     document.getElementById("water-img").style.color = '#999393';
     document.getElementById("food-img").style.color = '#999393';
-    document.getElementById("title").innerHTML = 'Steps Taken <span id="steptitle-annotation1">Distance Travelled</span>';
-    document.getElementById("subtitle").innerHTML = '<span id="stepsubtitle-annotation1" style="color: #05AD13;">3.2 miles</span>'
-    document.getElementById("pie-title").innerHTML = 'Today';
+	document.getElementById("pie-title").innerHTML = 'Today';
+	document.getElementById('chart-header-title').innerHTML = ' <p class="steps-ch-title" id="title"> Steps Taken </p>'
+	document.getElementById('total-value').innerHTML = '   <p class="steps-total-title" id="title"> Distance Travelled </p> <p class="steps-st-today subtitle" id="chart-steps-total-subtitle"> 3.2 miles </p>'
```
Comment (NavinVinayagam) [inline src/app/app.ts]: remove the unwanted white space in inner HTML

File: src/app/app.ts [hunk 5]
```diff
@@ -1297,13 +1303,12 @@ function sleepclick() {
     document.getElementById("sleep-img").style.color = '#FFFFFF';
     document.getElementById("water-img").style.color = '#999393';
     document.getElementById("food-img").style.color = '#999393';
-    document.getElementById("title").innerHTML = 'Sleep Tracker <span id="sleeptitle-annotation1">Daily Average</span><span id="sleeptitle-annotation2">Goal</span>';
-    document.getElementById("subtitle").innerHTML = '<span id="sleepsubtitle-annotation1">6.32 hrs</span><span id="sleepsubtitle-annotation2">7.5 hrs</span>'
-    document.getElementById("sleepsubtitle-annotation1").style.color = '#4526A6';
-    document.getElementById("sleepsubtitle-annotation2").style.color = '#4526A6';
-
+    document.getElementById('chart-header-title').innerHTML = '<p class="sleep-ch-title" id="title"> Sleep Tracker </p>'
```
Comment (NavinVinayagam) [inline src/app/app.ts]: Ensure whether do you all the attributes like paragraph tag, class name, ID

File: src/index.html [hunk 6]
```diff
@@ -8,106 +8,86 @@
...
 <body style=" font-family: 'Roboto', sans-serif;">
     <div id="container" style=" background: #FAFAFA;">
-        <div class="row" id="overall-title" style="cursor: default">
-            <p class="page-title">Health <b>Stat</b></p>
-            <img src="../../icons/Healthstat.png" style="padding-left: 1.5%; margin-top: 10px; float:left;">
-            <img src="../../icons/Profile_.png" style="padding-right: 1.5%; margin-top: 10px; float:right">
-            <p id="sub-title" style=" float: right">Hi, John Smith</p>
+        <!-- Appbar area -->
+        <div id="appbar">
+            <div class="appbar-content"><span class="e-avatar custom-avatar e-avatar-circle"><img src="../../icons/Healthstat.png"></span><p class="appbar-title"> Health <b>Stat</b> </p>
```
Comment (NavinVinayagam) [inline src/index.html]: Check whether appbar-content is necessary

File: src/index.html [hunk 7]
```diff
@@ -8,106 +8,86 @@
...
                             <div class="card-block" id="caloriescard" style="box-shadow: 0 3px 6px 2px rgba(178,30,195,0.30);">
-                                <div class="media">
-                                    <div class="media-body text-xs-left float-right" id="calories-value" style="max-width: 445px; background-image: linear-gradient(45deg, #F23B3B 0%, #EF9027 100%); border-radius: 4px">
-                                        <h3 class="card-value" id="calories-text" style="color: #FFFFFF">13,100</h3>
-                                        <div id="foodimg"><span class="card-text" id="calories-subtitle" style="color: #FFFFFF">Calories
-                                                Eaten</span><span id="food-img" class="icon-Food" style="font-size: 65px; margin-right: 3%; margin-bottom: 2% "></span>
-                                        </div>
+                                    <div class="media-body" id="calories-value" style="max-width: 445px; background-image: linear-gradient(45deg, #F23B3B 0%, #EF9027 100%); border-radius: 4px"><h3 class="card-value" id="calories-text" style="color: #FFFFFF">13,100</h3>
```
Comment (NavinVinayagam) [inline src/index.html]: Ensure proper alignment

File: src/styles/styles.css [hunk 8]
```diff
@@ -28,6 +28,34 @@
...
+.e-appbar.custom-appbar .appbar-title{
+    font-family: 'Roboto';
+    font-weight: 300;
+    font-size: 18px;
+    margin: 0;
+    color: #1A1A1A;
+    letter-spacing: 0;
+    display:inline-block;
+}
+.e-appbar.custom-appbar .appbar-subtitle{
```
Comment (NavinVinayagam) [inline src/styles/styles.css:51]: Try to club these 2 CSS

File: src/styles/styles.css [hunk 9]
```diff
@@ -84,31 +114,12 @@
...
-#overall-title {
-    background: #FFFFFF;
-    box-shadow: 0 2px 7px 0 rgba(0,0,0,0.16);
-    height: 60px;
-    margin-bottom: 20px
-}
 #container {
     padding-bottom: 20px;
     background: #FAFAFA !important;
+    box-shadow: 0 2px 7px 0 rgb(0 0 0 / 16%);
```
Comment (NavinVinayagam) [inline src/styles/styles.css:131]: Is this needed?

File: src/styles/styles.css [hunk 10]
```diff
@@ -775,215 +576,48 @@ text-align: center;
...
-        font-weight: 700;
-        font-size: 17px;
-        color: #DB4247;
-        letter-spacing: 0;
-        margin-right: 0% !important;
-        margin-top: -20px;
+@media only screen and (max-width:375px)
+{
+    #Break-text, #Lunch-text, #Dinner-text, #Snack-text {
+        font-size: 12px !important;
```
Comment (NavinVinayagam) [inline src/styles/styles.css]: Avoid using !important

File: src/styles/styles.css [hunk 11]
```diff
@@ -775,215 +576,48 @@ text-align: center;
...
-        color: #DB4247;
-        letter-spacing: 0;
-        margin-top: -20px;
-        margin-right: 4%;
+}
+
+@media only screen and (min-width: 213px) and (max-width: 514px) {
+    /* For portrait layouts only */
+    .e-avatar.custom-avatar{
+        font-size: 1.2rem;
```
Comment (NavinVinayagam) [inline src/styles/styles.css]: Why rem is used instead of px

**Review by NavinVinayagam — CHANGES_REQUESTED** [review]

File: src/index.html [hunk 12]
```diff
@@ -8,106 +8,109 @@
...
     <link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" />
     <!--style reference from node_modules/@syncfusion/ej2/-->
-    <link href="/styles/styles.css" rel="stylesheet" />
-
-    <script src="node_modules/systemjs/dist/system.src.js" type="text/javascript"></script>
+    <link href="./styles/styles.css" rel="stylesheet" />
+    <script src="/node_modules/systemjs/dist/system.src.js" type="text/javascript"></script>
     <script src="system.config.js" type="text/javascript"></script>
-
+    <link rel="stylesheet" href="/node_modules/@syncfusion/ej2-navigations/styles/material.css">
```
Comment (NavinVinayagam) [inline src/index.html]: Move this import state to styles.css,
avoid unnecessary alignment changes

**Review by NavinVinayagam — APPROVED** [review]

File: src/styles/styles.css (hunk 8 above)
Comment (manikandan-mariappan) [reply src/styles/styles.css:51]: Clubbed Common CSS into Single Blocks

File: src/styles/styles.css (hunk 9 above)
Comment (manikandan-mariappan) [reply src/styles/styles.css:131]: Yes box-shadow is needed for the outer shadow of the container. Previously it was loaded from external stylesheet.

File: src/app/app.ts (hunk 1 above)
Comment (manikandan-mariappan) [reply src/app/app.ts]: Appbar definition has been moved below the global variables

File: src/app/app.ts (hunk 2 above)
Comment (manikandan-mariappan) [reply src/app/app.ts]: Unnecessary removal of spaces has been avoided

File: src/app/app.ts (hunk 3 above)
Comment (manikandan-mariappan) [reply src/app/app.ts]: Style has applied inline to the element. 
Generic class names has been avoided instead simpler names are used

File: src/app/app.ts (hunk 4 above)
Comment (manikandan-mariappan) [reply src/app/app.ts]: Unwanted white spaces in inner HTML has been removed

File: src/app/app.ts (hunk 5 above)
Comment (manikandan-mariappan) [reply src/app/app.ts]: Ensured the need of tags and class names and modified it

File: src/index.html (hunk 6 above)
Comment (manikandan-mariappan) [reply src/index.html]: unnecessary class name appbar-content has been removed

File: src/index.html (hunk 7 above)
Comment (manikandan-mariappan) [reply src/index.html]: Alignments has been ensured

File: src/styles/styles.css (hunk 10 above)
Comment (manikandan-mariappan) [reply src/styles/styles.css]: Unnecessary !important tags has been removed

File: src/styles/styles.css (hunk 11 above)
Comment (manikandan-mariappan) [reply src/styles/styles.css]: converted rem into px.

File: src/index.html (hunk 12 above)
Comment (manikandan-mariappan) [reply src/index.html]: Moved imports into separate stylesheet.

File: src/app/app.ts [hunk 13]
```diff
@@ -31,9 +33,85 @@ let annotationpie2: AccumulationChart;
...
 
+let appbarObj = new AppBar({
+    cssClass: 'custom-appbar'
+});
+appbarObj.appendTo("#appbar");
+function InitializeSkeleton(): void {
+    document.querySelectorAll('.e-card').forEach((card: HTMLElement) => {
+        addClass([card], "hidden-element-size")
+    })
+    const cardElement: HTMLElement = document.querySelector('.e-card.hidden-element-size') as HTMLElement
```
Comment (NavinVinayagam) [inline src/app/app.ts]: Don't use the class "hidden-element-size" other than the intended purpose (To get the element size when the element )

File: src/app/app.ts [hunk 14]
```diff
@@ -31,9 +33,85 @@ let annotationpie2: AccumulationChart;
...
+    getCardData().then((data: any) => {
+        document.querySelectorAll('.card-skeleton').forEach((elem: HTMLElement) => elem.style.display = 'none');
+        (document.getElementById('line-chart-skeleton') as HTMLElement).style.display = 'none';
+        (document.getElementById('pie-chart-skeleton') as HTMLElement).style.display = 'none';
+        document.getElementById('calories-text').textContent = data['calories-eaten']
+        document.getElementById('steps-text').textContent = data['steps-taken']
+        document.getElementById('water-text').textContent = data['water-consumed']
+        document.getElementById('sleep-text').textContent = data['sleep-duration']
+        document.querySelectorAll('.e-card').forEach((elem: HTMLElement) => {
+            removeClass([elem], "hidden-element-size")
```
Comment (NavinVinayagam) [inline src/app/app.ts]: Remove the *hidden-element-size* class as soon as it's usage is over

File: src/app/app.ts [hunk 15]
```diff
@@ -31,9 +33,85 @@ let annotationpie2: AccumulationChart;
...
+        });
+        (document.getElementById('line') as HTMLElement).style.display = 'block';
+        linechartObj.refresh();
```
Comment (NavinVinayagam) [inline src/app/app.ts]: What is the need for refresh method here?

File: src/app/app.ts [hunk 16]
```diff
@@ -31,9 +33,85 @@ let annotationpie2: AccumulationChart;
 let annotation: boolean = true;
 let selectedpoint: boolean = false;
 export let category: string[] = [];
+InitializeSkeleton();
+loadCardComponent();
```
Comment (NavinVinayagam) [inline src/app/app.ts]: Why loadCardComponent method is called here?

File: src/app/app.ts [hunk 17]
```diff
@@ -31,9 +33,85 @@ let annotationpie2: AccumulationChart;
...
+        data['steps-taken'] = '52,100'
+        data['water-consumed'] = '38.7 ltr'
+        data['sleep-duration'] = '50 hr'
+        resolve(data)
+    }, 2000))
+}
+function loadCardComponent(): void {
+    getCardData().then((data: any) => {
+        document.querySelectorAll('.card-skeleton').forEach((elem: HTMLElement) => elem.style.display = 'none');
+        (document.getElementById('line-chart-skeleton') as HTMLElement).style.display = 'none';
```
Comment (NavinVinayagam) [inline src/app/app.ts]: Skeleton should be hidden only after updating the data

**Review by NavinVinayagam — CHANGES_REQUESTED** [review]

File: src/app/app.ts (hunk 13 above)
Comment (manikandan-mariappan) [reply src/app/app.ts]: Unnecessary usage of "hidden-element-size" has been avoided

File: src/app/app.ts (hunk 14 above)
Comment (manikandan-mariappan) [reply src/app/app.ts]: Removed the class after the usage got over

File: src/app/app.ts (hunk 15 above)
Comment (manikandan-mariappan) [reply src/app/app.ts]: call to refresh() method has been removed

File: src/app/app.ts (hunk 16 above)
Comment (manikandan-mariappan) [reply src/app/app.ts]: Modified the Method Structures as discussed

File: src/app/app.ts (hunk 17 above)
Comment (manikandan-mariappan) [reply src/app/app.ts]: Skeleton has been hidden after data loaded

File: src/index.html [hunk 18]
```diff
@@ -12,113 +12,128 @@
...
                 <div class="column-chart-area">
                     <div class="col-sm-12 col-md-4" id="pie-chart" style=" padding: 0px; height: 525px; ">
+                        <div id="pie-chart-skeleton"></div>
+                        <div id="pie-wrapper" style="display: none;">
                         <div class="chart-padding" id="pie">
                             <div id="donut" style="display: block; height: 530px;">
                                 <div class="col-xs-9 col-sm-9 col-md-9 col-lg-9" id="pie" style=" padding: 0px;">
-                                    <div id="pie-title" style=" padding-top: 20px; margin-bottom: 6px; cursor: default; margin-left: 20px; font-family: Roboto;  font-weight: 500; font-size: 20px; text-align: left; color: #1A1A1A">Sunday
-                                        Activity</div>
+                                    <div id="pie-title" style=" padding-top: 20px; margin-bottom: 6px; cursor: default; margin-left: 20px; font-family: Roboto;  font-weight: 500; font-size: 20px; text-align: left; color: #1A1A1A">Macro
```
Comment (NavinVinayagam) [inline src/index.html:134]: Why you have changed Sunday activity to macro nutrients?

Comment (manikandan-mariappan) [reply src/index.html:134]: The pie-title was not supposed to be "Sunday Activity". The correct format is "Macro Nutrients"

File: src/index.html [hunk 19]
```diff
@@ -12,108 +12,123 @@
...
     <link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" />
     <!--style reference from node_modules/@syncfusion/ej2/-->
-    <link href="/styles/styles.css" rel="stylesheet" />
+    
 
-    <script src="node_modules/systemjs/dist/system.src.js" type="text/javascript"></script>
+    <script src="/node_modules/systemjs/dist/system.src.js" type="text/javascript"></script>
     <script src="system.config.js" type="text/javascript"></script>
-
+    <link href="/node_modules/@syncfusion/ej2-base/styles/material.css" rel="stylesheet" />
```
Comment (NavinVinayagam) [inline src/index.html:15]: Move the style links to line 15

File: src/index.html [hunk 20]
```diff
@@ -12,108 +12,123 @@
         rel="stylesheet">
     <link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" />
     <!--style reference from node_modules/@syncfusion/ej2/-->
-    <link href="/styles/styles.css" rel="stylesheet" />
+    
 
-    <script src="node_modules/systemjs/dist/system.src.js" type="text/javascript"></script>
+    <script src="/node_modules/systemjs/dist/system.src.js" type="text/javascript"></script>
```
Comment (NavinVinayagam) [inline src/index.html]: why you have added /

File: src/index.html [hunk 21]
```diff
@@ -12,108 +12,123 @@
...
             <div class="row">
                 <div class="col-xs-6 col-xl-3 col-lg-3 col-md-3 col-sm-6 calories-valueactive" id="calories-column"
-                    style="padding-right: 0px; padding-left: 20px">
-                    <div class="card" id="calories" style="max-width: 445px; background-image: linear-gradient(45deg, #F23B3B 0%, #EF9027 100%)">
-                        <div class="card-body expense-card" id="calories-bg" style=" background-image: linear-gradient(45deg, #F23B3B 0%, #EF9027 100%); border-radius: 4px">
-                            <div class="card-block" id="caloriescard" style="box-shadow: 0 3px 6px 2px rgba(178,30,195,0.30);">
-                                <div class="media">
-                                    <div class="media-body text-xs-left float-right" id="calories-value" style="max-width: 445px; background-image: linear-gradient(45deg, #F23B3B 0%, #EF9027 100%); border-radius: 4px">
-                                        <h3 class="card-value" id="calories-text" style="color: #FFFFFF">13,100</h3>
+                    style="padding-right: 0px; padding-left: 20px; margin-bottom: 20px;">
```
Comment (NavinVinayagam) [inline src/index.html]: confirm the usage for margin-bottom

**Review by NavinVinayagam — CHANGES_REQUESTED** [review]

File: src/index.html (hunk 20 above)
Comment (manikandan-mariappan) [reply src/index.html]: It is added, so that the file is referenced correctly

File: src/index.html (hunk 19 above)
Comment (manikandan-mariappan) [reply src/index.html:15]: Moved to Line 15

File: src/index.html (hunk 21 above)
Comment (manikandan-mariappan) [reply src/index.html]: added an helper class and moved the margin styles to external sheet instead of inline.

**Review by NavinVinayagam — APPROVED** [review]

**Review by NavinVinayagam — APPROVED** [review]

**Review by syncsiva — APPROVED** [review]

**Pull request merged**